### DISCUSSION TITLE
Ability to get a Provider instance with react native wallet connect

### DISCUSSION THREAD

**Discussion by intergalacticspacehighway** [WalletConnect/walletconnect-monorepo#700] [Developer Support]
Thanks for the library!

I am using it with react-native and following this [documentation](https://docs.walletconnect.com/quick-start/dapps/react-native). Is there a way to get a Provider object with `useWalletConnect` hook?

I referred [web3 Provider](https://docs.walletconnect.com/quick-start/dapps/react-native) docs, but this appears to be web specific and also it won't be connected to my wallet.

For my usecase, I need the Provider instance as some other SDKs depend on it for signing messages. e.g. [Biconomy](https://docs.biconomy.io/products/enable-gasless-transactions/network-agnostic-transactions/custom-approach#4.-sign-and-send-meta-transactions)

I am happy to submit a PR if someone can provide some direction. Thanks!

**Comment by paul2234** [comment]
I'm trying to do the same thing. But there does not seem to be a standard solution (or if there is, it is poorly documented).

I'm pretty new to react native and wallet connect, so take what I say with a grain of salt.

Here is what I'm trying to do.
1. Instantiate web3 provider according to normal web3js. I'm using an HTTP provider with my infura node ex. `const web3 = React.useMemo(
    () => new Web3(new Web3.providers.HttpProvider("https://rinkeby.infura.io/v3/<infura node id>")),
    []
  );`. I think the React.useMemo is a bit redundant. I did have some use for useMemo in earlier iterations of my code and no longer have any dependencies, but I've left the syntax as is.
2. With the provider, instantiate a contract. ex `const contract = new web3.eth.Contract(contractAbi,contractAddress);`
3. Get the ABI from the method & parameters. ex. `const methodABI = contract.methods.myMethod([param1, param2, param3,...).encodeABI()`. ([web3js documentation on this method])(https://web3js.readthedocs.io/en/v1.7.1/web3-eth-contract.html#methods-mymethod-encodeabi)
5. With the connector instance we've created with useWalletConnect (ex. `const connector = useWalletConnect();`), sign a transaction with the encoded ABI. example (details from [wallet connect documentation](https://docs.walletconnect.com/json-rpc-api-methods/ethereum#example-parameters-1)):
`connector.signTransaction({
    "from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
    "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
    "data": methodABI,
    "gas": "0x76c0", // 30400
    "gasPrice": "0x9184e72a000", // 10000000000000
    "value": "0x9184e72a", // 2441406250
    "nonce": "0x117" // 279
});`

Will let you know if this works.

**Reply by paul2234** [reply]
Quick update: while reading through issues on the `create-react-native-dapp` repo (recommended by walletconnect), I did find that the connector object has an rpcUrl attribute, so you can instantiate a web3 instance with:

`const web3 = new Web3(connector.rpcUrl);`. I still have yet to fully test this, but figured I'd share.

**Reply by intergalacticspacehighway** [reply]
@paul2234 the `rpcUrl` seems to be undefined for me. Did you find any alternative?
